### ISSUE TITLE
On addon-docs DocsPage, PropType.shape renders as unknown

### ISSUE THREAD

**Issue by ghost** [storybookjs/storybook#9399]
I have a component with the following proptypes:

```
Credits.propTypes = {
  areas: PropTypes.arrayOf(
    PropTypes.shape({
      position: PropTypes.string.isRequired,
      names: PropTypes.arrayOf(PropTypes.string).isRequired,
    }),
  ).isRequired,
};
```

But on the DocsPage I am getting `unknown[]` instead of the `shape` definition inside the array:

<img width="1027" alt="Screen Shot 2020-01-12 at 19 09 20" src="https://user-images.githubusercontent.com/47044668/72226429-e0f4c100-356f-11ea-8e30-c0cadf62adff.png">

**Comment by shilman** [comment]
Reproduced here: https://github.com/storybookjs/storybook/commit/8063b5291c97a95504f780977b16030ef7172c7a

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
